AirSkyBoat/AirSkyBoat
# Fixes Summoner BPs

**_I affirm:_**

## What does this pull request do?
Complete summoner rework and refractor based off of retail captured data from: https://docs.google.com/spreadsheets/d/1YBoveP-weMdidrirY-vPDzHyxbEI2ryECINlfCnFkLI/edit#gid=557476280

**Physical Bloodpact Damage Calc**
(Weapon D + fSTR + WSC) x fTP x pdif x pdt
1) You perform this calculation from left to right
2) WSC for all formulae is subject to alpha correction
3) Any multipliers are summed and converted to an interger(Floor)
4) Multi-hit bp offhits use an FTP of 1, no FTP transfer in era
5) Physical Bloodpacts are absorbed by utsusemi, but Magical Bloodpacts aren't

**Magical Blood Pact Damage Calc**
((Lvl+2 + WSC) x fTP + dstat) x Magic Burst bonus x resist x day/ weather bonus x  MAB/MDB x mdt
1) You can apply the points below to physical bp calcs by subbing dstat for fSTR
2) dstat is usually avatar int - target int
3) When dstat is positive it is multiplied by 1.5, when it is negative is is just the raw value
4) There is no upper limit of dstat, but there is a lower limit of -65
5) Magic Burst bonus: 1.3 Base + multi-step skillchain (if 2 then .5) + enemy resistance 0~1.5
6) Somnolence and Nether Blast do not use WSC or dstat calcs

**Physical + Magical Bloodpact Formula**
(FTP * pDIF * D) + (MAB/MDB) * (FTP) * pDIF * D)
1) Burning Strike and Flaming Crush equations only


## Steps to test these changes
- Go to any mob
- check its difficulty. If it is too weak the pdif should be capped at 2.0 and not be hitting them for 5k damage
- Damage should be normalized throughout the board


Fixes https://github.com/AirSkyBoat/AirSkyBoat/issues/1737
